named.go

```diff
@@ -224,21 +224,28 @@ func bindStruct(bindType int, query string, arg interface{}, m *reflectx.Mapper)
 	return bound, arglist, nil
 }
 
-var valueBracketReg = regexp.MustCompile(`\([^(]*.[^(]\)\s*$`)
+var valueBracketReg = regexp.MustCompile(`VALUES\s+(\([^(]*.[^(]\))`)
 
 func fixBound(bound string, loop int) string {
-	loc := valueBracketReg.FindStringIndex(bound)
-	if len(loc) != 2 {
+	loc := valueBracketReg.FindAllStringSubmatchIndex(bound, -1)
+	// Either no VALUES () found or more than one found??
+	if len(loc) != 1 {
+		return bound
+	}
+	// defensive guard. loc should be len 4 representing the starting and
+	// ending index for the whole regex match and the starting + ending
+	// index for the single inside group
+	if len(loc[0]) != 4 {
```

Comment:
If `loc[0] != 4`, that indicates that the regex was changed and either the submatcher was removed or another submatcher was added.